helpers/calendar/analytics_helper.py

```diff
@@ -85,9 +85,12 @@ def get_all_events_in_a_room(self, calendar_id, min_limit, max_limit):
             - min_limit, max_limit(Time range)
         """
         service = Credentials.set_api_credentials(self)
-        events_result = service.events().list(
-            calendarId=calendar_id, timeMin=min_limit, timeMax=max_limit,
-            singleEvents=True, orderBy='startTime').execute()
+        try:
+            events_result = service.events().list(
+                calendarId=calendar_id, timeMin=min_limit, timeMax=max_limit,
+                singleEvents=True, orderBy='startTime').execute()
+        except Exception:
+            raise GraphQLError("Resource not found")
```

Comment:
Can this be done by catching the specific error instead of a broad error.

Reply:
@C3real-kill3r please respond to caveinn's comment. Is there a specific error you are trying to catch?

Reply:
Sorry about this. The error I was trying to catch was an HTTPError. But when I declared it explicitly, it was failing. I also tried to catch `urllib2.HTTPError` handle it, and if it's Error 404, raise that exception; this too failed to work. So the only option I had left after extensive research was to use a bare except (is a flake8 error) or use the exception class since all built-in, non-system-exiting exceptions are derived from this class; and this got it to work, unless you have any suggestions?

Reply:
It seems you have already tried out all the suggestions I had.